### PR TITLE
API cleanup

### PR DESCRIPTION
**Pull request by 9prady9**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,9 @@ PROJECT(FORGE)
 
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
 
-OPTION(BUILD_EXAMPLES "Build Examples" ON)
+OPTION(BUILD_EXAMPLES "Build Examples" OFF)
+OPTION(USE_SYSTEM_GLM "Use system GLM" OFF)
+OPTION(USE_SYSTEM_FREETYPE "Use system GLM" OFF)
```
Comment (shehzan10) [inline CMakeLists.txt]: Copy paste error

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -91,12 +93,11 @@ ADD_LIBRARY(forge SHARED
...
-                      ${GLFW_LIBRARY}
-                      ${GLEWmx_LIBRARY}
-                      ${OPENGL_gl_LIBRARY}
-                      ${OPENGL_glu_LIBRARY}
-                      ${FREETYPE_LIBRARIES}
-                      ${FONTCONFIG_LIBRARY}
-                      )
+    ${FONTCONFIG_LIBRARY}
+    ${FREETYPE_LIBRARIES}
+    ${OPENGL_gl_LIBRARY}
```
Comment (shehzan10) [inline src/CMakeLists.txt]: what happened to glu?

Comment (9prady9) [inline src/CMakeLists.txt]: Removed gluErrorString from the code and we are not using glu anywhere apart from that one. We can write our own error string function based on error codes.

File: CMakeLists.txt (hunk 1 above)
Comment (9prady9) [inline CMakeLists.txt]: Corrected it.

**Pull request merged**